### DISCUSSION TITLE
A nvda specific pypi repo for addon update and download

### DISCUSSION THREAD

**Discussion by larry801** [nvaccess/nvda#12517] [Ideas]
we can publish nvda python code as a pypi package.
All new addons can be developed as a package depending on nvda root package.

**Comment by onyx-and-iris** [comment]
Is this a suggestion that you will do it? Or are you asking if you can? I have the same question. I noticed the controller client is covered by LGPL. Does this mean I can push the (unmodified) controller client files to pypi as part of a python package? Basically, a thin wrapper around its C functions.

If it's allowed I'm happy to do it, for my own purposes but also because it might be useful for someone else.

**Reply by seanbudd** [reply]
There's no issue doing this according to the project GPL licence

**Comment by seanbudd** [comment]
>we can publish nvda python code as a pypi package.
All new addons can be developed as a package depending on nvda root package.

NVDA is currently not packaged/structured appropriately for this
